models/src/main/protobuf/CasperMessage.proto

```diff
@@ -106,22 +106,30 @@ message BlockHashMessageProto {
   bytes  blockCreator = 2;
 }
 
+message DeployChain {
+  repeated bytes deploys       = 1 [(scalapb.field).collection_type="collection.immutable.List"];
+}
+
+message StateMetadata {
+  repeated DeployChain proposed         = 1 [(scalapb.field).collection_type="collection.immutable.List"];
+  repeated DeployChain acceptedSet      = 3 [(scalapb.field).collection_type="collection.immutable.List"];
+  repeated DeployChain rejectedSet      = 4 [(scalapb.field).collection_type="collection.immutable.List"];
+}
+
 message BlockMetadataInternal {
   // This message in mapped to a different Scala class because of protobuf's inability to create map<bytes, int64> for
   // bonds.
   option (scalapb.message).type = "coop.rchain.models.BlockMetadata";
 
   bytes blockHash                       = 1;
-  repeated bytes parents                = 2 [(scalapb.field).collection_type="collection.immutable.List"];
+  bytes postStateHash                   = 2;
```

Comment:
I am very curious what is the intention to remove parents?

Reply:
oh. It is just meta data.But still curious

Reply:
We do not need parents - only justifications. Parents were computed by Estimator with ranking, to compute main parent, which we do not have now. Some version of parents might be an optimisation for traversal of the DAG to not traverse multiple times through the same message if justification A sees justification B, but this is not done now, and will require some more thinking with regards to distance from the tips.

But long story short - we do not need a notion of parents.

Reply:
I am not sure I am following.

Reply:
What is the purpose of having parents?

Reply:
If its not clear about distance - we need to find highest messages, and this highest can be detected by measuring Stream length when traversing. https://github.com/rchain/rchain/pull/3512/commits/7481cb404bbc3c1019ba3d6790451585785d5dcb#diff-8ca81baa47046b2e5e687927b9ad064e12dd12af17415e3003adc662d45e979dR48-R57